File: messageBlock.js
// 😉 Credits: Anuken
// 👨‍💻 Converted from https://github.com/Anuken/Mindustry/blob/master/core/src/mindustry/world/blocks/logic/MessageBlock.java

class MessageBlock {
    constructor(){
        this.message = "";
        this.maxTextLength = 220;
        this.maxNewLines = 24;
    }
    config(text){
        if (text.length > this.maxTextLength){
            return; //no. -Anuken
        }
        this.message = "";
        text = text.trim();
        let count = 0;
        for(let i = 0; i < text.length; i++){
            let c = text.charAt(i);
            if(c == '\n'){
                if(count++ <= this.maxNewLines){
                    this.message += '\n';
                }
            }
            else{
                this.message += c;
            }
        }
    }
}

export {MessageBlock};

File: messageBlock.cpp
// 😉 Credits: Anuken
// 👨‍💻 Converted from https://github.com/Anuken/Mindustry/blob/master/core/src/mindustry/world/blocks/logic/MessageBlock.java

#include <string>
using namespace std;

string trim(string s){
    string whitespace = " \n\t\r";
    int first = 0;
    for(int i = 0; i < s.length();i++){
        if(string(whitespace).find(s[i]) || 0){
            first = i;
            break;
        }
    }
    if(first == s.length()-1){
        return "";
    }

    int last = s.length() - 1;
    for(int i = s.length() - 1; i >= 0;i--){
        if(string(whitespace).find(s[i]) || 0){
            last = i;
            break;
        }
    }

    return s[first,last];
}

class MessageBlock {
    public:
        string message = "";
        int maxTextLength = 220;
        int maxNewLines = 24;
        MessageBlock(){}
        void config(string text){
            if (text.length() > maxTextLength){
                return; //no. -Anuken
            }
            message = "";
            text = text;
            int count = 0;
            for(int i = 0; i < text.length(); i++){
                int c = text[i];
                if(c == '\n'){
                    if(count++ <= maxNewLines){
                        message += '\n';
                    }
                }
                else{
                    message += c;
                }
        }
    }
};